test/e2e/e2eutil.go

```diff
@@ -29,15 +29,28 @@ func RegisterTypesWithFramework(newTypes ...runtime.Object) error {
 	return nil
 }
 
-func CreateOrUpdateMongoDB(mdb *mdbv1.MongoDB, ctx *f.TestCtx) error {
-	err := f.Global.Client.Get(context.TODO(), types.NamespacedName{Name: mdb.Name, Namespace: mdb.Namespace}, &mdbv1.MongoDB{})
-	if err != nil {
-		if errors.IsNotFound(err) {
-			return f.Global.Client.Create(context.TODO(), mdb, &f.CleanupOptions{TestContext: ctx})
+// UpdateMongoDBResource applies the provided function to the most recent version of the MongoDB resource
+// and retries when there are conflicts
+func UpdateMongoDBResource(original *mdbv1.MongoDB, updateFunc func(*mdbv1.MongoDB)) error {
+	for i := 0; i < 3; i++ {
+		err := f.Global.Client.Get(context.TODO(), types.NamespacedName{Name: original.Name, Namespace: original.Namespace}, original)
+		if err != nil {
+			return err
+		}
+
+		updateFunc(original)
```

Comment:
The original object is passed to the function, then it is being fetched from the API, then it is modified in place, then it is updated...

Unfortunatelly this means that the same object could be updated multiple times, and the updated object will be fetched from the API...

The "original" `original` object will have to be saved so every iteration is independant from the next.

Reply:
resolved in person